src/catalog/src/builtin.rs

```diff
@@ -3278,6 +3278,49 @@ GROUP BY object_id, collection_timestamp",
     access: vec![PUBLIC_SELECT],
 });
 
+pub static MZ_RECENT_STORAGE_USAGE: Lazy<BuiltinView> = Lazy::new(|| BuiltinView {
+    name: "mz_recent_storage_usage",
+    schema: MZ_INTERNAL_SCHEMA,
+    oid: oid::VIEW_MZ_RECENT_STORAGE_USAGE_OID,
+    column_defs: Some("object_id, size_bytes"),
+    sql: "
+WITH
+
+recent_storage_usage_by_shard AS (
+    SELECT shard_id, size_bytes, collection_timestamp
+    FROM mz_internal.mz_storage_usage_by_shard
+    -- Restricting to the last 6 hours makes it feasible to index the view.
+    WHERE collection_timestamp + '6 hours' >= mz_now()
+),
+
+most_recent_collection_timestamp_by_shard AS (
+    SELECT shard_id, max(collection_timestamp) AS collection_timestamp
+    FROM recent_storage_usage_by_shard
+    GROUP BY shard_id
+)
+
+SELECT
+    object_id,
+    sum(size_bytes)::uint8
+FROM
+    mz_internal.mz_storage_shards
+    LEFT JOIN most_recent_collection_timestamp_by_shard
+        ON mz_storage_shards.shard_id = most_recent_collection_timestamp_by_shard.shard_id
+    LEFT JOIN recent_storage_usage_by_shard
+        ON mz_storage_shards.shard_id = recent_storage_usage_by_shard.shard_id
+        AND most_recent_collection_timestamp_by_shard.collection_timestamp = recent_storage_usage_by_shard.collection_timestamp
+GROUP BY object_id",
+    access: vec![PUBLIC_SELECT],
+});
+
+pub static MZ_RECENT_STORAGE_USAGE_IND: Lazy<BuiltinIndex> = Lazy::new(|| BuiltinIndex {
+    name: "mz_recent_storage_usage_ind",
+    schema: MZ_INTERNAL_SCHEMA,
+    oid: oid::INDEX_MZ_RECENT_STORAGE_USAGE_IND_OID,
+    sql: "IN CLUSTER mz_catalog_server ON mz_internal.mz_recent_storage_usage (object_id, size_bytes)",
```

Comment:
What benefit does adding `size_bytes` to the index add? Wouldn't it make lookups by object id not possible? I thought the non-indexed columns were still stored alongside the index?